### ISSUE TITLE
Installing Hopswork 3.2.0 pep517 issues

### ISSUE THREAD

**Issue by llctrautmann** [fastavro/fastavro#705]
## Issue

I have mighty issues installing hopsworks on my local machine with poetry or pip. Two packages seem to be required with specific version numbers that cannot be installed at the moment:

__Confluent-kafka and fastavro__

## Error Logs
In both cases the following error code is returned. 

`1 error generated. error: command '/usr/bin/clang' failed with exit code 1`
  
`Note: This error originates from the build backend, and is likely not a problem with poetry but with confluent-kafka (1.9.0) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "confluent-kafka (==1.9.0)"'.`

## Attempted Remedy
I have so far tried to install the packages separately but this runs into version issues with hopsworks 3.2.0.  

`Because hsfs (3.2.0) depends on confluent-kafka (<=1.9.0) and no versions of hsfs match >3.2.0,<3.3.0, hsfs (>=3.2.0,<3.3.0) requires confluent-kafka (<=1.9.0). And because hopsworks (3.2.0) depends on hsfs (>=3.2.0,<3.3.0), hopsworks (3.2.0) requires confluent-kafka (<=1.9.0). So, because <PROJECT> depends on both confluent-kafka (2.1.0) and hopsworks (3.2.0), version solving failed.
`
The same error occurs for __fastavro__. 

Machine: 
Hardware: MacBook M2 (2023), Apple M2 Max, 64 GB Memory, Ventura 13.4

If more logs are required I will added them as soon as possible.

**Comment by scottbelden** [comment]
Since this was closed I'm assuming you were able to figure out a solution?

**Comment by llctrautmann** [comment]
Yes, sorry, I just mixed up where I wanted to raise the issue. There was an issue with/is still an issue with fastavro not compiling with cython3 (see: https://github.com/fastavro/fastavro/issues/701) that seems to affect the installation with pip/poetry and that in turn broke the hopswork install. But from what I got from hopswork and the issue here, the problem is fixed on your end and hopsworks is also working on a patch. But thanks for checking!
